Makefile

```diff
@@ -9,33 +9,36 @@ _MOCHA=./node_modules/.bin/_mocha
 UGLIFY=./node_modules/.bin/uglifyjs
 NYC=./node_modules/.bin/nyc
 
-.PHONY: test test-all-nodejs all clean coverage
+.PHONY: test test-all-nodejs coverage lib/defs.js
 
-all: lib/defs.js
+error:
+	@echo "Please choose one of the following targets: test, test-all-nodejs, coverage, lib/defs.js"
+	@exit 1
 
-clean:
-	rm lib/defs.js bin/amqp-rabbitmq-0.9.1.json
-	rm -rf ./coverage
-
-lib/defs.js: $(UGLIFY) bin/generate-defs.js bin/amqp-rabbitmq-0.9.1.json
-	(cd bin; node ./generate-defs.js > ../lib/defs.js)
-	$(UGLIFY) ./lib/defs.js -o ./lib/defs.js \
-		-c 'sequences=false' --comments \
-		-b 'indent-level=2' 2>&1 | (grep -v 'WARN' || true)
-
-test: lib/defs.js
+test:
 	$(MOCHA) --check-leaks -u tdd --exit test/
 
-test-all-nodejs: lib/defs.js
+test-all-nodejs:
 	for v in $(NODEJS_VERSIONS); \
 		do echo "-- Node version $$v --"; \
 		nave use $$v $(MOCHA) -u tdd --exit -R progress test; \
 		done
 
-coverage: $(NYC) lib/defs.js
-	$(NYC) --reporter=lcov --reporter=text $(_MOCHA) -u tdd -R progress test/
+coverage: $(NYC)
+	$(NYC) --clean --reporter=lcov --reporter=text $(_MOCHA) -u tdd --exit -R progress test/
```

Comment:
Instead of the clean target deleting the .coverage directory, I thought it better to have nyc do it. 

Mocha was hanging so I added the --exit argument

Reply:
What is causing Mocha to hang? It's generally a good idea to avoid `--exit`, because otherwise we can miss problems with resource disposal.

Reply:
I'll see if I can work it out. Most likely a connection to the broker not being closed in one of the tests.

Reply:
There are loads of test where the connection isn't being closed. Really this should be done on both success and failure, so it will take a while to fix them all. I'll create a separate issue for it and leave the --exit for now.